pyhoma/client.py

```diff
@@ -226,6 +231,42 @@ async def nexity_login(self) -> str:
 
             return token["token"]
 
+    @backoff.on_exception(
+        backoff.expo,
+        (NotAuthenticatedException, ServerDisconnectedError),
+        max_tries=2,
+        on_backoff=relogin,
+    )
+    async def get_setup(self, refresh: bool = False) -> Setup:
+        """
+        Get all data about the connected user setup
+            -> gateways data (serial number, activation state, ...): <gateways/gateway>
+            -> setup location: <location>
+            -> house places (rooms and floors): <place>
+            -> setup devices: <devices>
+
+        A gateway may be in different modes (mode) regarding to the activated functions (functions).
+        A house may be composed of several floors and rooms. The house, floors and rooms are viewed as a place.
+        Devices in the house are grouped by type called uiClass. Each device has an associated widget.
+        The widget is used to control or to know the device state, whatever the device protocol (controllable): IO, RTS, X10, ... .
+        A device can be either an actuator (type=1) or a sensor (type=2).
+        Data of one or several devices can be also get by setting the device(s) url as request parameter.
+
+        Per-session rate-limit : 1 calls per 1d period for this particular operation (bulk-load)
+        """
+        if self.setup and not refresh:
+            return self.setup
+
+        response = await self.__get("setup")
+        setup = Setup(**humps.decamelize(response))
+
+        # Cache response
+        self.setup = setup
+        self.gateways = setup.gateways
+        self.devices = setup.devices
```

Comment:
Here we override the `self.devices` set within `get_devices`. It’s on purpose?

Reply:
When we retrieve everything from /setup, we also get the devices. This way we save it to the cache, so when you call get_devices, it doesn't need to do an API call again for the same info.

Reply:
Perfect if it’s exactly the same response!